### PR TITLE
Codebase update - please see diff

### PR DESCRIPTION
**Pull request by reidobandedo**
I've done my best to put good commit msgs if you take a look. basically, i went thru marriott's game.js file and made sure we weren't missing anything blatant that we'd need later (ie. reset methods for our entities when we start another round of the game). @suchk & @Jazzmatazz - i stubbed out some platform code that you might merge into, so take a look (it's mainly just comments). feel free to overwrite it's contents.

### CODE REVIEW

File: index.html [hunk 1]
```diff
@@ -7,26 +7,36 @@
...
-    -->
-
     <script type="text/javascript" src="./js/background.js"></script>
     <script type="text/javascript" src="./js/assetmanager.js"></script>
     <script type="text/javascript" src="./js/main.js"></script>
 </head>
 <body>
-    <canvas id="gameWorld" width="800" height="600"></canvas>
+    <div style="position: absolute; left: 15px; top: 30px; width: 400px; height: 30px; background-color: rgba(0,0,0,0);
+<<<<<<< HEAD
```
Comment (ddh) [inline index.html]: You have Merge comments in here still (<<<<<<HEAD)

File: js/goat.js [hunk 2]
```diff
@@ -45,23 +43,46 @@ function Goat(game) {
...
     this.game = game;
     this.ctx = game.ctx;
 
+    this.boundingbox = new BoundingBox(this.x, this.y, this.standAnimation.frameWidth, this.standAnimation.frameHeight);
+
+    // TODO: figure out this value later!
+    this.jumpHeight = 200;
+
+    // TODO: add anymore boolean flags needed for debugging
+    this.boxes = false; // used to draw bounding boxes around goat as it goes thru its different animations
```
Comment (ddh) [inline js/goat.js]: rename `this.boxes` to `this.debug` possibly? - done

File: js/goat.js [hunk 3]
```diff
@@ -45,23 +43,46 @@ function Goat(game) {
...
     Entity.call(this, game, 0, 650); // changed value from 400
 }
 
 Goat.prototype = new Entity();
 Goat.prototype.constructor = Goat;
 
-Goat.prototype.setBoundingCircle = function (circle) {
-    this.circle = circle;
+Goat.prototype.reset = function () {
+    this.standing   = true;
```
Comment (ddh) [inline js/goat.js:72]: Just curious; can a goat ever have more than one "state"?

I think for simplicity, let's just say no for now. But one scenario I've thought of is when a goat is falling but is also stunned.

File: js/playgame.js [hunk 4]
```diff
@@ -0,0 +1,39 @@
+// This code is based on Chris Marriott's Unicorn game found here:
```
Comment (ddh) [inline js/playgame.js:1]: I take it that PlayGame is responsible for all game logistics involving scoring, time, and any housekeeping outside of actually playing the game?

Right now, it just controls the button to start and restart rounds of the game and keeps track of the number of rounds played.

File: index.html (hunk 1 above)
Comment (reidobandedo) [inline index.html]: done

**Pull request merged**